include/PetriEngine/PQL/XMLPrinter.h

```diff
@@ -28,16 +28,17 @@ namespace PetriEngine {
         class XMLPrinter : public Visitor {
         public:
             XMLPrinter(std::ostream& os, uint32_t init_tabs = 0, uint32_t tab_size = 2, bool print_newlines = true, bool token_count = false) :
-                os(os), token_count(token_count), tabs(init_tabs), tab_size(tab_size),
-                        print_newlines(print_newlines) {}
+                os(os), token_count(token_count), tabs(init_tabs),
+                        print_newlines(print_newlines) {
+                single_tab = std::string(tab_size, ' ');
+            }
 
         protected:
             std::ostream& os;
             const bool token_count;
             uint32_t tabs;
-            const uint32_t tab_size;
             const bool print_newlines;
-            static constexpr char single_tab = ' ';
```

Comment:
Why is this changed back into a string?

Reply:
The tab size is set by `tab_size` in the constructor, so a single tab can be more than one space. Before the tab_size was unused. It is more convenient to write `<< single_tab << "something else"` compared to something like `<< repeat(single_tab, tab_size) << "something else"`.